internal/http/handler/mcp_handler.go

```
@@ -95,6 +112,161 @@ func (h *MCPHandler) HandlePost(c echo.Context) error {
+// populateSDKStandardHeaders supplies the required 2026 protocol routing
+// headers when a client has sent a valid JSON-RPC request but omitted them.
+// Existing values are validated rather than overwritten so a mismatched
+// header cannot be used to route a different method or tool.
+func populateSDKStandardHeaders(request *http.Request, payload []byte) error {
+	var envelope struct {
+		JSONRPC string          `json:"jsonrpc"`
+		Method  string          `json:"method"`
+		Params  json.RawMessage `json:"params"`
+	}
+	if err := json.Unmarshal(payload, &envelope); err != nil || envelope.JSONRPC != "2.0" || strings.TrimSpace(envelope.Method) == "" {
+		return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 192) but it returns nil

(nilerr)